custom_components/ai_thermostat/climate.py

```diff
@@ -70,823 +67,747 @@
+				elif self.heating_active_pre_window_opened:
+					self._hvac_mode = HVAC_MODE_HEAT
+				
+				try:
+					remapped_states = convert_outbound_states(self, self._hvac_mode)
+					converted_hvac_mode = remapped_states.get('system_mode')
+					current_heating_setpoint = remapped_states.get('current_temperature')
+					has_real_mode = True if self.hass.states.get(self.heater_entity_id).attributes.get('system_mode') is not None else False
+					calibration = remapped_states.get('calibration')
+					
+					# if off do nothing
+					if self.closed_window_triggered or self._hvac_mode == HVAC_MODE_OFF:
+						if has_real_mode:
+							if self.hass.states.get(self.heater_entity_id).attributes.get('system_mode') == HVAC_MODE_OFF:
+								self.ignore_states = False
+								return
+						if self.calibration_type == 1:
+							if float(self.hass.states.get(self.heater_entity_id).attributes.get('current_heating_setpoint')) <= 5.0:
+								self.ignore_states = False
+								return
+					
+					# Only send the local_temperature_calibration if not instantly following
+					do_calibration = False
+					if (datetime.now() > (self.last_calibration + timedelta(seconds=20))):
+						do_calibration = True
+						self.last_calibration = datetime.now()
+					
+					_LOGGER.debug(
+							"ai_thermostat triggered states > window open: %s night mode: %s Mode: %s set: %s has_mode: %s Calibration: %s - send: %s set_temp: %s cur_temp: %s Model: %s Calibration "
+							"type: %s winter: %s TRV: %s",
+							self.window_open,
+							self.night_status,
+							converted_hvac_mode,
+							self._hvac_mode,
+							has_real_mode,
+							calibration,
+							do_calibration,
+							current_heating_setpoint,
+							self._cur_temp,
+							self.model,
+							self.calibration_type,
+							is_cold,
+							self.hass.states.get(self.heater_entity_id).attributes.get('device').get('friendlyName')
+					)
+					
+					if self.calibration_type == 1 and float(self.hass.states.get(self.heater_entity_id).attributes.get('current_heating_setpoint')) != float(calibration):
+						data = {ATTR_ENTITY_ID: self.heater_entity_id, "temperature": float(calibration)}
+						await self.hass.services.async_call('climate', SERVICE_SET_TEMPERATURE, data)
+					
+					if self.calibration_type == 0 and not self.window_open and self.hass.states.get(self.heater_entity_id).attributes.get('current_heating_setpoint') != float(
+							current_heating_setpoint) and converted_hvac_mode != HVAC_MODE_OFF and float(current_heating_setpoint) != 5.0 and is_cold:
+						data = {ATTR_ENTITY_ID: self.heater_entity_id, "temperature": float(current_heating_setpoint)}
+						await self.hass.services.async_call('climate', SERVICE_SET_TEMPERATURE, data)
+									
+					# Calibration stuff
+					if self.calibration_type == 0 and not self.window_open and do_calibration:
+						local_calibration_entity = self.heater_entity_id.replace('climate.', 'number.')+'_local_temperature_calibration'
```

Comment:
We need a way to get this entity from the climate entity, it should be related to it, but i didn't find out how to get related entities. @RubenKelevra

Reply:
I commented on that on the other mention (: